### PR TITLE
Use Parcel application bundler

### PR DESCRIPTION
**Pull request by karantan**
Ref: https://github.com/allo-media/elm-kitchen/issues/29

### CODE REVIEW

File: skeleton/package.json [hunk 1]
```diff
@@ -5,25 +5,19 @@
   "author": "You",
   "license": "MIT",
   "scripts": {
-    "build": "npm run copy-assets && elm make src/Main.elm --optimize --output=build/app.js && npm run optimize",
-    "copy-assets": "rimraf build && mkdir build && cp -Lr public/* build",
-    "debug": "elm-live src/Main.elm --port=3000 --dir=public/ -- --output=public/app.js --debug",
-    "deploy": "npm run build && gh-pages --dist build/",
-    "live": "elm-live src/Main.elm --port=3000 --dir=public/ -- --output=public/app.js",
-    "optimize": "uglifyjs build/app.js -c $npm_package_uglify_params | uglifyjs -m --output=build/app.js",
```
Comment (n1k0) [inline skeleton/package.json]: Would be nice to keep/adapt the `deploy` and `optimize` commands.

File: skeleton/package.json [hunk 2]
```diff
@@ -5,25 +5,19 @@
...
-    "start": "npm run live",
+    "build": "npx parcel build src/index.html",
+    "start": "npx parcel src/index.html",
     "test": "elm-test"
   },
-  "uglify": {
-    "params": "pure_funcs=\"F2,F3,F4,F5,F6,F7,F8,F9,A2,A3,A4,A5,A6,A7,A8,A9\",pure_getters,keep_fargs=false,unsafe_comps,unsafe"
-  },
   "devDependencies": {
+    "cssnano": "^4.1.10",
```
Comment (n1k0) [inline skeleton/package.json]: I'm wondering why this has to be installed explicitely, isn't it a dependency of Parcel itself?

File: skeleton/package.json [hunk 3]
```diff
@@ -5,25 +5,19 @@
...
     "elm": "^0.19.0-no-deps",
     "elm-format": "^0.8.1",
-    "elm-live": "^3.4.1",
+    "elm-hot": "^1.1.4",
```
Comment (n1k0) [inline skeleton/package.json]: Same question

File: skeleton/src/app.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+app.ports.saveStore.subscribe((rawStore) => {
+    localStorage[storeKey] = rawStore;
+});
+
+// Ensure session is refreshed when it changes in another tab/window
+window.addEventListener("storage", (event) => {
+    if (event.storageArea === localStorage && event.key === storeKey) {
+    app.ports.storeChanged.send(event.newValue);
+    }
+}, false);
```
Comment (n1k0) [inline skeleton/src/app.js]: Nits:

- 2 spaces indentation + fix formatting
- Add trailing newline

**Review by n1k0 — COMMENTED** [review]
Looks pretty rad, thanks! 

A question I have though, it seems the Elm debugger is enabled by default, which is most often unwanted in my own experience; Is there any way to disable it by default, while maybe offering another command to enable it explicitly, as we had before?

**Comment by karantan** [comment]
> A question I have though, it seems the Elm debugger is enabled by default, which is most often unwanted in my own experience; Is there any way to disable it by default, while maybe offering another command to enable it explicitly, as we had before?

I couldn't find this option with parcel. It looks like it is sending the `--debug` flag by default (Ref: https://github.com/parcel-bundler/parcel/issues/2104).

File: skeleton/package.json (hunk 1 above)
Comment (karantan) [reply skeleton/package.json]: Parcel only has `build` and `serve` (and `watch`) commands. If I understand correctly, parcel passes `--optimize` flag automatically. See https://github.com/parcel-bundler/parcel/issues/2913

File: skeleton/package.json (hunk 2 above)
Comment (karantan) [reply skeleton/package.json]: Yes, Parcel needs it.

File: skeleton/package.json (hunk 3 above)
Comment (karantan) [reply skeleton/package.json]: Same answer :)

**Comment by karantan** [comment]
@n1k0 I've updated the PR based on your suggestions.

**Comment by n1k0** [comment]
@karantan thanks for the update, I'm still puzzled with enforcing the Elm debug mode with no obvious way to disable it. From what I've read in Parcel issues, it will be addressed in Parcel 2 with *custom transformers* but it's yet to be released, so maybe we should just wait for that specific release?

File: skeleton/package.json (hunk 1 above)
Comment (n1k0) [reply skeleton/package.json]: Maybe just keep the deploy command at least, then ;)

**Comment by karantan** [comment]
> @karantan thanks for the update, I'm still puzzled with enforcing the Elm debug mode with no obvious way to disable it. From what I've read in Parcel issues, it will be addressed in Parcel 2 with _custom transformers_ but it's yet to be released, so maybe we should just wait for that specific release?

ok

**Comment by merwan** [comment]
Project is now deprecated #50

**Pull request closed without merging**